### PR TITLE
Add release notes for v2.7.0

### LINKED ISSUE #6967: Release Version 2.7.0
This is a component issue for 2.7.0.
Coming from [opensearch-build#3230](https://github.com/opensearch-project/opensearch-build/issues/3230). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help.
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `__2.x__` branch to `__2.7.0__` for Min/Core, and `__2.7.0.0__` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [ ] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.7.0`.

### CI/CD

- [x] All code changes for `2.7.0` are complete.
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.7.0).

### Pre-Release

- [x] Update to the `2.7` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.7.0).
- [x] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `2.7.0` have been merged.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.
- [x] Verify the release date mentioned in release notes is correct and matches actual release date.

### Post Release

- [x] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [x] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/component_release_template.md).
- [x] Conduct a retrospective, and publish its results.

### PR DESCRIPTION
**Pull request by Rishikesh1159**
### Description
This PR adds release notes for the v2.7.0 release and cleans up the CHANGELOG to only capture unreleased changes.

### Issues Resolved
#6967 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Rishikesh1159** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/14263/
> * **CommitID:** [7d332cc](https://github.com/opensearch-project/OpenSearch/commit/7d332cce715b8713c9b9974bd413fad0801a7b04)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

will wait for this PR : https://github.com/opensearch-project/OpenSearch/pull/7238 to merge in and then retry the gradle check

**Review by dreamer-89 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,72 +75,23 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
-- Bump `org.codehaus.jettison:jettison` from 1.5.3 to 1.5.4 ([#6878](https://github.com/opensearch-project/OpenSearch/pull/6878))
-- Add `com.github.luben:zstd-jni:1.5.5-1` ([#3577](https://github.com/opensearch-project/OpenSearch/pull/3577))
-- Bump: Netty from 4.1.90.Final to 4.1.91.Final , ASM 9.4 to ASM 9.5, ByteBuddy 1.14.2 to 1.14.3 ([#6981](https://github.com/opensearch-project/OpenSearch/pull/6981))
-- Bump `com.azure:azure-storage-blob` from 12.15.0 to 12.21.1
 
 ### Changed
-- Require MediaType in Strings.toString API ([#6009](https://github.com/opensearch-project/OpenSearch/pull/6009))
-- [Refactor] XContent base classes from xcontent to core library ([#5902](https://github.com/opensearch-project/OpenSearch/pull/5902))
-- Changed `opensearch-env` to respect already set `OPENSEARCH_HOME` environment variable ([#6956](https://github.com/opensearch-project/OpenSearch/pull/6956/))
-- Added a new field type: flat_object ([#6507](https://github.com/opensearch-project/OpenSearch/pull/6507))
```
Comment (macohen) [inline CHANGELOG.md:130]: 6507 is a brand new feature. can this be moved to added?

Comment (andrross) [reply CHANGELOG.md:130]: Sure thing! @macohen want to open a PR to move the line in the release notes file?

Comment (macohen) [reply CHANGELOG.md:130]: Yes. Yes I will!

**Pull request merged**